packages/fx-core/src/core/middleware/projectMigrator.ts

```
@@ -0,0 +1,427 @@
+  const envConfig = await fs.readJson(path.join(fx, "env.default.json"));
+  if (envConfig[ResourcePlugins.Bot]) {
+    delete envConfig[ResourcePlugins.Bot];
+    envConfig[ResourcePlugins.Bot] = { wayToRegisterBot: "create-new" };
+  }
+  await fs.writeFile(path.join(fx, "new.env.default.json"), JSON.stringify(envConfig, null, 4));
+}
+
+async function generateArmTempaltesFiles(ctx: CoreHookContext) {
+  const fakeCtx: CoreHookContext = { arguments: ctx.arguments };
+  const inputs = ctx.arguments[ctx.arguments.length - 1] as Inputs;
+  const core = ctx.self as FxCore;
+
+  const fx = path.join(inputs.projectPath as string, `.${ConfigFolderName}`);
+  const fxConfig = path.join(fx, InputConfigsFolderName);
+  const templateAzure = path.join(inputs.projectPath as string, "templates", "azure");
+  await fs.ensureDir(fx);
+  await fs.ensureDir(fxConfig);
+  await fs.ensureDir(templateAzure);
+  // load local settings.json
+  const loadRes = await loadProjectSettings(inputs);
+  if (loadRes.isErr()) {
+    throw ProjectSettingError();
+  }
+  const [projectSettings, projectIdMissing] = loadRes.value;
+  fakeCtx.projectSettings = projectSettings;
+  fakeCtx.projectIdMissing = projectIdMissing;
+
+  // load envinfo env.default.json
+  const targetEnvName = "default";
+  const result = await loadSolutionContext(
+    core.tools,
+    inputs,
+    fakeCtx.projectSettings,
+    fakeCtx.projectIdMissing,
+    targetEnvName,
+    inputs.ignoreEnvInfo
+  );
+  if (result.isErr()) {
+    throw SolutionConfigError();
+  }
+  fakeCtx.solutionContext = result.value;
+  // generate bicep files.
+  try {
+    await generateArmTemplate(fakeCtx.solutionContext);
+  } catch (error) {
+    return error;
+  }
+  if (await checkFileExist(path.join(templateAzure, "parameters.template.json"))) {
+    await fs.move(
+      path.join(templateAzure, "parameters.template.json"),
+      path.join(fxConfig, "azure.parameters.dev.json")
+    );
+  } else {
+    const parameterObject = {
```

we may need to report an error instead generating an empty parameter.dev.json, since provision with an empty parameter.dev.json will always fail.

---

ok!